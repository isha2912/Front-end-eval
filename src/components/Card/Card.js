/* eslint-disable react/prop-types */
import React from 'react';
import Like from '../Like/Like';
import './Card.scss';

const Card = (({ eachSong }) => (

  <div className="card">
    <img className="song-image" src={eachSong.albumArtUrl} alt="img" />
    <div className="container-top">
      <div className="container-bottom">
        <div>
          {eachSong.name}
          {' '}
        </div>
        <div>{eachSong.artist.name}</div>
      </div>
      <Like eachSong={eachSong} />
    </div>
  </div>
));

export default Card;
