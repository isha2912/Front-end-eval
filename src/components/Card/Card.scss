.card{
    display:flex;
    flex-direction: column;
    width: 10rem;
    background-color:#DFDFDF;
    margin: 1rem;

}
.song-image{
    height:10rem;
    width:8rem;
    border-radius: 100%;
    border:solid black
}
.heart-button{
    height: 0.5rem;
    width: 0.5rem;
    border: collapse;
}
.container-top{
    display: flex;
}