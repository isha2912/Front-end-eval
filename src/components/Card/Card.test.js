/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import Card from './Card';

describe(Card.name, () => {
  const mockSong = {
    eachSong: {
      name: 'yariyan',
      artist: {
        name: 'Arijit singh',
      },
      albumArtUrl: 'url',
      id: '9',
    },
    index: 1,
  };
  test('Should match snapshot', () => {
    const { container } = render(<Card {...mockSong} />);
    expect(container).toMatchSnapshot();
  });
  test('Should display song names', () => {
    render(<Card {...mockSong} />);
    screen.getByText('yariyan');
  });

  test('should display artist', () => {
    render(<Card {...mockSong} />);
    screen.getByText('Arijit singh');
  });
});
