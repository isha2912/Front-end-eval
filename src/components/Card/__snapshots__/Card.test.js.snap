// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="img"
      class="song-image"
      src="url"
    />
    <div
      class="container-top"
    >
      <div
        class="container-bottom"
      >
        <div
          class="song-name"
        >
          yariyan
           
        </div>
        <div>
          <b>
            Arijit singh
          </b>
        </div>
      </div>
      <button
        type="button"
      >
        <img
          alt="heart"
          src="./assets/heart-gray.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
